pkg/controller.v1beta1/trial/trial_controller.go

```diff
@@ -174,6 +175,12 @@ type ReconcileTrial struct {
 	collector *trialutil.TrialsCollector
 }
 
+// Map which contains number of requeuing for each trial if observation logs are not available
+// That is needed if Job is succeeded but metrics are not reported yet
+// Key = Trial name, value = requeue count
+var trialRequeueCount = make(map[string]int)
```

Comment:
What will happen if the controller is restarted?

Reply:
@gaocegege If metrics are not available it tries to reconcile Trial for `maxRequeueCount` again after the restart.

Reply:
OK, we are sure that there is no side effect right?

Reply:
Only one problem that I can see that `trialRequeueCount` is waisting some container resources, is that fine?

Reply:
SGTM